### PR TITLE
Remove trivial warnings, unused imports, and update Newtonsoft.Json

### PR DESCRIPTION
**Pull request by 1nf0rmagician**
This PR removes all unsued imports in the solution. It inserts comments or removes the xml comment warnings where the comments where trivial and it makes the necessary Newtonsoft update

### CODE REVIEW

File: src/Moryx.AbstractionLayer.Products.Endpoints/ProductPermissions.cs [hunk 1]
```diff
@@ -1,15 +1,53 @@
...
+
+        /// <summary>
+        /// Permission for all actions related to viewing one or multiple product types
+        /// </summary>
         public const string CanViewTypes = _prefix + "CanViewTypes";
+
+        /// <summary>
+        /// Permission for all CRUD actions related to recipes
+        /// </summary>
         public const string CanCreateAndEditRecipes = _prefix + "CanCreateAndEditRecipes";
```
Comment (MarisaGoergen) [inline src/Moryx.AbstractionLayer.Products.Endpoints/ProductPermissions.cs:21]: can this also delete recipes? If yes, should this be renamed?

File: src/Moryx.AbstractionLayer.TestTools/Moryx.AbstractionLayer.TestTools.csproj [hunk 2]
```diff
@@ -8,14 +8,9 @@
...
   </PropertyGroup>
 
-  <ItemGroup>
-	<PackageReference Include="Microsoft.EntityFrameworkCore" />
-  </ItemGroup>
-
   <ItemGroup>
 	<ProjectReference Include="..\Moryx.Model.Sqlite\Moryx.Model.Sqlite.csproj" />
 	<ProjectReference Include="..\Moryx.AbstractionLayer\Moryx.AbstractionLayer.csproj" />
-	<ProjectReference Include="..\Moryx.TestTools.UnitTest\Moryx.TestTools.UnitTest.csproj" />
```
Comment (MarisaGoergen) [inline src/Moryx.AbstractionLayer.TestTools/Moryx.AbstractionLayer.TestTools.csproj:18]: why is that not needed anymore? Was that ever needed?

File: src/Moryx.Products.Model/Entities/PartLinkEntity.cs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) 2023, Phoenix Contact GmbH & Co. KG
 // Licensed under the Apache License, Version 2.0
 
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
```
Comment (MarisaGoergen) [inline src/Moryx.Products.Model/Entities/PartLinkEntity.cs:4]: Are entities not that important and that is why you disabled the warnings?

File: src/Moryx.Resources.Management/ModuleController/ModuleConsole.cs [hunk 4]
```diff
@@ -31,12 +31,6 @@ internal class ModuleConsole : IServerModuleConsole
 
         #endregion
 
-        #region Fields
-
-        private IResourceInitializer[] _initializers;
```
Comment (MarisaGoergen) [inline src/Moryx.Resources.Management/ModuleController/ModuleConsole.cs:36]: what was that for?

File: src/Moryx.AbstractionLayer.TestTools/Moryx.AbstractionLayer.TestTools.csproj (hunk 2 above)
Comment (1nf0rmagician) [reply src/Moryx.AbstractionLayer.TestTools/Moryx.AbstractionLayer.TestTools.csproj:18]: It was imported in multiple projects but the automatic detection of unused imports from visual studio detected it multiple times as not use. Maybe @Toxantron can tell us what it was originally used/imported for 🤔

File: src/Moryx.Resources.Management/ModuleController/ModuleConsole.cs (hunk 4 above)
Comment (1nf0rmagician) [reply src/Moryx.Resources.Management/ModuleController/ModuleConsole.cs:36]: @Toxantron, maybe you can answer that from the top of your head. But it was a private field that was not used in the class, that's why I removed it 😊

File: src/Moryx.AbstractionLayer.Products.Endpoints/ProductPermissions.cs (hunk 1 above)
Comment (1nf0rmagician) [reply src/Moryx.AbstractionLayer.Products.Endpoints/ProductPermissions.cs:21]: nope it is also not necessary to read recipes, I changed the comment

File: src/Moryx.Runtime.Kernel/Modules/Components/MissingFacadeException.cs [hunk 5]
```diff
@@ -49,6 +49,7 @@ public MissingFacadeException(string moduleName, string propName, Type facadeTyp
         /// <param name="context">The System.Runtime.Serialization.StreamingContext that contains contextual
         /// information about the source or destination.</param>
         /// <exception cref="System.ArgumentNullException">The info parameter is a null reference</exception>
+        [Obsolete("Override of an obsolete method.")]
```
Comment (Toxantron) [inline src/Moryx.Runtime.Kernel/Modules/Components/MissingFacadeException.cs:52]: Just delete it

File: src/Moryx.Runtime/Modules/Management/HealthStateException.cs [hunk 6]
```diff
@@ -58,6 +58,7 @@ public HealthStateException(ServerModuleState current)
         /// <param name="context">The System.Runtime.Serialization.StreamingContext that contains contextual
         /// information about the source or destination.</param>
         /// <exception cref="System.ArgumentNullException">The info parameter is a null reference</exception>
+        [Obsolete("Override of an obsolete method.")]
```
Comment (Toxantron) [inline src/Moryx.Runtime/Modules/Management/HealthStateException.cs:61]: Same

File: src/Moryx/Configuration/InvalidConfigException.cs [hunk 7]
```diff
@@ -51,6 +51,7 @@ public InvalidConfigException(object faultyEntry, string propertyFailure)
         /// <param name="context">The System.Runtime.Serialization.StreamingContext that contains contextual
         /// information about the source or destination.</param>
         /// <exception cref="System.ArgumentNullException">The info parameter is a null reference</exception>
+        [Obsolete("Override of an obsolete method.")]
```
Comment (Toxantron) [inline src/Moryx/Configuration/InvalidConfigException.cs:54]: Same

**Comment by Toxantron** [comment]
@1nf0rmagician **ping** to wrap this up?

**Comment by 1nf0rmagician** [comment]
@Toxantron thanks for the reminder, I removed the obsolete methods **and** the constructors which where calling an obsolete base constructor in `Exception`. With your approval we can merge this 😊

**Review by Toxantron — APPROVED** [review]

**Pull request merged**